quint/src/parsing/ToIrListener.ts

```diff
@@ -1270,13 +1241,39 @@ export class ToIrListener implements QuintListener {
   // pop a row
   private popRow(): Row {
     // the user has specified a row
-    const tp = this.rowStack.pop()
-    return tp!
+    return this.rowStack.pop() ?? { kind: 'empty' }
   }
 
-  private undefinedDef(ctx: any): QuintEx {
-    const id = this.getId(ctx)
-    return { id, kind: 'bool', value: true }
+  private undefinedExpr(ctx: any): () => QuintEx {
+    return () => {
+      const id = this.getId(ctx)
+      return { id, kind: 'bool', value: true }
+    }
+  }
+
+  private undefinedDeclaration(ctx: any): () => QuintDeclaration {
+    return () => {
+      const id = this.getId(ctx)
+      return { id, kind: 'assume', name: '_', assumption: this.undefinedExpr(ctx)() }
```

Comment:
Can we name it something like

```suggestion
      return { id, kind: 'assume', name: `_undefinedDecl${id}`, assumption: this.undefinedExpr(ctx)() }
```